module/zfs/arc.c

```diff
@@ -10849,3 +11002,6 @@ ZFS_MODULE_PARAM(zfs_arc, zfs_arc_, evict_batch_limit, UINT, ZMOD_RW,
 
 ZFS_MODULE_PARAM(zfs_arc, zfs_arc_, prune_task_threads, INT, ZMOD_RW,
 	"Number of arc_prune threads");
+
+ZFS_MODULE_PARAM(zfs_arc, zfs_arc_, evict_threads, UINT, ZMOD_RW,
+	"Maximum number of arc_evict threads");
```

Comment:
For read-only tunables there is `ZMOD_RD`, which should allow initial setting, but not later modification.  After that you would not need `zfs_arc_evict_threads_live`.

Reply:
It's a good idea, I'll do that. I wasn't sure if that's acceptable to use ZMOD_RD in this case. Thank you for the clarifying that.

Reply:
Hmmm, I could set it to ZMOD_RD and get rid of `zfs_arc_evict_threads_live`, but then consider the following scenario that may occur with the new design:

- The user keeps the zfs_arc_evict_threads parameter at 0.
- The user loads ZFS
- The user checks the value of zfs_arc_evict_threads
- Instead of `0`, ZFS reports the number of live threads on the system resulting from auto-scaling

Wouldn't that be confusing?

Reply:
Using `ZMOD_RD` here is a good idea anyway, but I am not sure if getting rid of `zfs_arc_evict_threads_live` is desirable.

Reply:
> Wouldn't that be confusing?

I could argue it both ways: it may be confusing, or it may be informative.  Would accept it either way.